packages/meditrak-server/src/apiV2/export/exportSurveyResponses/exportResponsesToFile.js

```diff
@@ -0,0 +1,303 @@
+
+    // Set up the left columns with info about the questions
+    questionsForExport.forEach((question, questionIndex) => {
+      const questionInfo = infoColumnKeys.map(columnKey => question[columnKey]);
+      const exportRow = preQuestionRowCount + questionIndex;
+      exportData[exportRow] = questionInfo;
+    });
+
+    // Add the answers on the right columns to the exportData
+    const questionIdToIndex = questionsForExport.reduce(
+      (dict, question, index) => ({ ...dict, [question.id]: index }),
+      {},
+    );
+    const responseIdToIndex = surveyResponses.reduce(
+      (dict, response, index) => ({ ...dict, [response.id]: index }),
+      {},
+    );
+    answers.forEach(answer => {
+      const surveyResponseIndex = responseIdToIndex[answer['survey_response.id']];
+      const questionIndex = questionIdToIndex[answer['question.id']];
+      const exportRow = preQuestionRowCount + questionIndex;
+      const exportColumn = infoColumnKeys.length + surveyResponseIndex;
+      exportData[exportRow][exportColumn] = answer?.text || '';
+    });
+
+    if (easyReadingMode) {
+      return addExportedDateAndOriginAtTheSheetBottom(exportData, timeZone);
+    }
+
+    return exportData;
+  };
+
+  const exportResponses = async (surveyResponses, entitiesById, survey, questions) => {
+    const exportData = await getExportDataForResponses(
+      surveyResponses,
+      entitiesById,
+      survey,
+      questions,
+    );
+    addDataToSheet(survey.name, exportData);
+  };
+
+  /** Main body of the function below this point, everything above is helper functions */
+
+  const accessBySurveyIndex = await Promise.all(surveys.map(checkAccessToSurvey));
+  const surveysWithoutAccess = surveys.filter((s, i) => !accessBySurveyIndex[i]);
+  surveysWithoutAccess.forEach(survey => {
+    const exportData = [[`You do not have export access to ${survey.name}`]];
+    addDataToSheet(survey.name, exportData);
+  });
+
+  const surveysWithAccess = surveys.filter((s, i) => accessBySurveyIndex[i]);
+  for (const survey of surveysWithAccess) {
+    // Get the current set of questions, in the order they appear in the survey
+    const questions = await findQuestionsInSurvey(models, survey.id);
+
+    if (surveyResponse) {
+      const entity = await models.entity.findById(surveyResponse.entity_id);
+      await exportResponses([surveyResponse], { [entity.id]: entity }, survey, questions);
+    } else {
+      const surveyResponses = await findResponsesForSurvey(survey);
+      const entitiesById = keyBy(entities, 'id');
+      if (surveyResponses.length > MAX_RESPONSES_PER_FILE) {
+        for (const batchOfResponses of chunk(surveyResponses, MAX_RESPONSES_PER_FILE)) {
+          await exportResponses(batchOfResponses, entitiesById, survey, questions);
+          saveCurrentWorkbook();
+        }
+      } else {
+        await exportResponses(surveyResponses, entitiesById, survey, questions);
+      }
```

Comment:
I prefer to keep it here, because otherwise you have the saving of the workbook happening inside that function as well as in this outer one (in the case where no batching is done), which feels a little like mixing responsibilities/confusing to me

Reply:
Good point! In that case, could we rename `exportResponses` to something else (eg `addResponsesToSheet`)?

Reply:
Oh yeah that's better!